### DISCUSSION TITLE
Install of local modified tap ignores formula diff?

### DISCUSSION THREAD

**Discussion by rmfuhrer** [Homebrew/discussions#5558] [Writing Formulae/Casks] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.3.16-6-g92ddc02
ORIGIN: https://github.com/Homebrew/brew
HEAD: 92ddc02005e1f1a64cdf754ddc887ad3f921dce2
Last commit: 7 hours ago
Core tap HEAD: 556b6b5e0d82e20bbb3b9e3fd33926e613fc052c
Core tap last commit: 27 hours ago
Core tap JSON: 17 Aug 01:03 UTC
Core cask tap JSON: 17 Aug 01:03 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: vi
HOMEBREW_GITHUB_API_TOKEN: set
HOMEBREW_MAKE_JOBS: 24
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 3.3.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.4_1/bin/ruby
CPU: 24-core 64-bit arm_blizzard_avalanche
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 8.6.0 => /usr/bin/curl
macOS: 14.5-arm64
CLT: 15.0.3.9.4
Xcode: 15.4
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Warning: Some taps are not on the default git origin branch and may not receive
updates. If this is a surprise to you, check out the default branch with:
  git -C $(brew --repo rmfuhrer/local-dovecot) checkout
```

### Description of issue

I need to tweak the formula for Dovecot to add 'depends_on "mysql"' to build in support for the MySQL database.

FWIW, I've been using Dovecot for several years from Homebrew with this basic setup. This is just supposed to be a re-build of what I already had working, after my previous Mac died due to a power surge.

My steps:

```
$ brew tap-new $USER/local-dovecot
$ brew tap --force homebrew/core
$ brew extract --version=2.3.21 dovecot $USER/local-dovecot
$ vi /opt/homebrew/Library/Taps/$USER//homebrew-local-dovecot/Formula/dovecot@2.3.21.rb    # Add "depends_on 'mysql'"
$ brew install -s $USER/local-dovecot/dovecot@2.3.21
```

Dovecot builds without error, and seems healthy in most respects, but sadly lacks the mysql driver, as evidenced by the output from "dovecot --build-options":

```
$ dovecot --build-options
Build options: ioloop=kqueue notify=kqueue openssl io_block_size=8192
SQL drivers: sqlite
Passdb: checkpassword pam passwd passwd-file sql
Userdb: checkpassword passwd prefetch passwd-file sql
$
```

Also, Dovecot's unable to spin up its auth service, which is configured to use mysql, and complains that it can't find the mysql driver.

I have MySQL v9.0.1 installed via Homebrew, and it's running properly.

I believe the above recipe worked in the past, but unfortunately I didn't document all of the steps in detail. Possibly there was some state on the old Mac that the new one doesn't have (the old one was Intel while the new one is Apple Silicon).

IIUC, "brew install -v -d..." would give me a chance to inspect the build directory's state, but only if the build fails, and as of right now, it's succeeding, so I'm not sure how to peek under the hood to see what compile flags ("-I", "-L", etc.) are getting used.

Any advice?

**Comment by Bo98** [comment, marked as answer]
Check `~/Library/Logs/Homebrew/dovecot` for build logs, including config.log that you may find the most useful.

**Reply by rmfuhrer** [reply]
Thanks! I wasn't aware the build/install logs were saved anywhere.

That said, they disappeared shortly after I looked at them. I can only assume that's intentional, but what's the retention policy?

Anyway, the logs helped me figure out that it was pilot error - I had forgotten to add 1 of the 2 diffs to the formula. <facepalm/>

All good now!
